### ISSUE TITLE
ParamSpec `_set_default` called from `ParamSpec.__new__` raises Attribute Error

### ISSUE THREAD

**Issue by ptmcg** [python/typing_extensions#405]
This code worked in Python 3.12, now fails in 3.13.0b1:
```python
from typing_extensions import ParamSpec

FactoryParamSpec = ParamSpec('FactoryParamSpec')
```
with this traceback:
```
Traceback (most recent call last):
  File "C:\Users\ptmcg\AppData\Roaming\JetBrains\PyCharm2024.1\scratches\scratch_3.py", line 4, in <module>
    FactoryParamSpec = ParamSpec('FactoryParamSpec')
  File "C:\Users\ptmcg\venvs\logmerger_3.13\Lib\site-packages\typing_extensions.py", line 1512, in __new__
    _set_default(paramspec, default)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ptmcg\venvs\logmerger_3.13\Lib\site-packages\typing_extensions.py", line 1375, in _set_default
    type_param.__default__ = None
    ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: attribute '__default__' of 'typing.ParamSpec' objects is not writable
```

**Comment by JelleZijlstra** [comment]
This should be fixed in the release candidate which we're planning to release on Thursday. Could you check with typing_extensions 4.12rc1?

**Comment by ptmcg** [comment]
Fixed in 4.12.0rc1. Thanks for jumping on this so quickly!
